File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/billing/shared/dto/service-item-details.dto.ts
/*
  Note: This is a replica of ServerDto: ServiceItemDetails_DTO -> inside DanpheEMR.Services.Billing.DTO

  Created: 16Mar'23--Sud
*/

export class ServiceItemDetails_DTO {
  public ServiceItemId: number = 0;
  public PriceCategoryId: number = 0;
  public SchemeId: number = 0;
  public ItemCode: string = null
  public ItemName: string = "";
  public ServiceDepartmentId: number = 0
  public ServiceDepartmentName: string = null;
  public Price: number = 0;
  public IsTaxApplicable: boolean = false;
  public IsDiscountApplicable: boolean = false;
  public DiscountPercent: number = 0;
  public DiscountAmount: number = 0;
  public IsPriceChangeAllowed: boolean = false;
  public IsZeroPriceAllowed: boolean = false;
  public HasAdditionalBillingItems: boolean = false;
  public IsDoctorMandatory: boolean = false;
  public IsCoPayment: boolean = false;
  public CoPayCashPercent: number = 0;
  public CoPayCreditPercent: number = 0;
  public IntegrationItemId: number = 0;
  public IntegrationName: string = null;
  public DisplaySequence: number = null;
  public IsPackageBilling: boolean = false;
  public PerformerId: number = null;
  public DefaultDoctorList: string = null;
}

File: Code/Websites/DanpheEMR/Services/Billing/DTO/ServiceItemDetails_DTO.cs
using System;

namespace DanpheEMR.Services.Billing.DTO
{
    /// <summary>
    /// Gives Price, Copayment, Discount details of the ServiceItem for Current PriceCategory and Scheme.
    /// </summary>
    public class ServiceItemDetails_DTO
    {
        public int ServiceItemId { get; set; }
        public int PriceCategoryId { get; set; }
        public int SchemeId { get; set; }
        public string ItemCode { get; set; }
        public string ItemName { get; set; }
        public int ServiceDepartmentId { get; set; }
        public string ServiceDepartmentName { get; set; }
        public decimal Price { get; set; }
        public bool IsTaxApplicable { get; set; }
        public bool IsDiscountApplicable { get; set; }
        public decimal DiscountPercent { get; set; }
        public bool IsPriceChangeAllowed { get; set; }
        public bool IsZeroPriceAllowed { get; set; }
        public bool HasAdditionalBillingItems { get; set; }
        public bool IsDoctorMandatory { get; set; }
        public bool IsCoPayment { get; set; }
        public decimal CoPayCashPercent { get; set; }
        public decimal CoPayCreditPercent { get; set; }
        public int IntegrationItemId { get; set; }
        public string IntegrationName { get; set; }
        public int? DisplaySequence { get; set; }
        public string DefaultDoctorList { get; set; }
    }
}
